### PR TITLE
Ruby auth add user creds

### LINKED ISSUE #4: ADC Support for User Refresh Tokens
The initial implementation of Application Default Credentials is missing one of the 3 credential types: User Refresh Tokens. This is basically the end result of a 3LO flow, and is an important part of the tool integration, as the Cloud SDK writes out this form of credential in the well-known-file location.

The file format to support is this:

{  
  "type": "authorized_user",     
  "client_id": "kflc91.apps.googleusercontent.com",     
  "client_secret": "1/olgReg3YaBQqxm6T",
  "refresh_token": "2/fFAGRNJru1FTz70BzhT3Zg"     
}
The "type" value is to differentiate it from a service account. The other 3 values are needed to get refresh tokens back. This is a common OAuth2 scenario, so there is likely to be an existing object that already supports getting access tokens from these values. This file could be in either the well-known file location or the environment variable location.

For reference, the Java implementation of these features is here:
https://github.com/google/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java
https://github.com/google/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

### PR DESCRIPTION
**Pull request by tbetbetbe**
Closes #4

### CODE REVIEW

**Comment by tbetbetbe** [comment]
@anthmgoogle @vsubramani. PTAL

File: lib/googleauth.rb [hunk 1]
```diff
@@ -40,6 +45,28 @@ module Auth
...
+        clz.new(scope, StringIO.new(MultiJson.dump(json_key)))
+      end
+
+      # Reads the input json and determines which creds class to use.
+      def self.determine_creds_class(json_key_io)
+        json_key = MultiJson.load(json_key_io.read)
+        fail "the json is missing the #{key} field" unless json_key.key?('type')
+        svc_account = json_key['type'] == 'service_account'
+        return json_key, ServiceAccountCredentials if svc_account
+        [json_key, UserRefreshCredentials]
```
Comment (anthmgoogle) [inline lib/googleauth.rb]: There may eventually be a 3rd type. Recommend checking the type for user auth and failing with an errir if the type is missing or has any other value.

**Comment by anthmgoogle** [comment]
Looks great. Just one small comment on the JSON parsing.

File: lib/googleauth.rb (hunk 1 above)
Comment (tbetbetbe) [inline lib/googleauth.rb]: Done

**Pull request merged**